Update tickets

gbirke opened the pull request:
Add command to update tickets with a new owner.

gbirke committed:
Add OTRS connector class

Wrapper around a guzzle web request.

gbirke committed:
Add Ticket update use case

gbirke committed:
Simplify ticket update use case input

Instead of using the whole ExtractedData model we just use a reader.

gbirke committed:
Fix request options for OTRS connector

POST request header and body must be separated.
Content type header must be set.

gbirke committed:
Add response processing to OtrsConnector

Add error checking to OtrsConnector and return response data.

gbirke committed:
Catch exception in UpdateTicketsUseCaseTest

Use case now catches and logs OtrsConnectorExceptions

gbirke committed:
Use strings for ticket numbers when updating

Since they are very long and not specified as numbers in the remote database, do not convert ticket numbers to int when updating ticket

gbirke committed:
Improve Ticket Update

Log ticket number when an error occurs.
Fox OTRS connector test - body is a JSON-encoded string that must be decoded.
Fix CS violations.

gbirke committed:
Add Ticket-Update CLI command

gbirke committed:
Update documentation

JeroenDeDauw commented on cli/UpdateTicketsCommand.php:
```diff
@@ -0,0 +1,55 @@
+<?php
+declare( strict_types = 1 );
```
Y u ranomly not put the newline between those two things?

![ocd](https://cloud.githubusercontent.com/assets/146040/22591652/eee164ce-ea15-11e6-8dac-d1e548cb9acc.png)

JeroenDeDauw commented on cli/UpdateTicketsCommand.php:
```diff
@@ -0,0 +1,55 @@
+	/**
+	 * UpdateTicketsCommand constructor.
+	 * @param UpdateTicketsUseCase $updateTicketsUseCase
+	 */
```
Clutter docblock

JeroenDeDauw commented on cli/UpdateTicketsCommand.php:
```diff
@@ -0,0 +1,55 @@
+	}
+
+	protected function configure()
+	{
```
Bracket positioning also not consistent

JeroenDeDauw commented on src/DataAccess/TicketNumberReader.php:
```diff
@@ -0,0 +1,27 @@
+		$inputFile->setCsvControl( ';', '"' );
+		$inputFile->setFlags( \SplFileObject::READ_CSV | \SplFileObject::READ_AHEAD | \SplFileObject::SKIP_EMPTY
+			| \SplFileObject::DROP_NEW_LINE );
+		parent::__construct( $inputFile );
```
This modification of something passed by-ref in the constructor is kinda unexpected. If another service also gets the same object and modifies it differently things go badly.

Since this class binds tightly to the SplFileObject anyway, perhaps that thing can be constructed in the constructor here? Looking at the users that is problematic for the nice in-memory unit test... 

This problem is coming from one object having both the contents and the way how that contents is read. Perhaps the API does not force that though? (I can't ssh on this network so hard to check stuff)

gbirke committed:
Use file path for TicketNumberReader

This addresses
https://github.com/wmde/otrs-address-extractor/pull/2#discussion_r99330871

gbirke committed:
Add cli path to code style check

wiese commented on cli/UpdateTicketsCommand.php:
```diff
@@ -0,0 +1,50 @@
+			->setName( 'update-tickets' )
+			->setDescription( 'Update tickets with new owner' )
+			->setHelp( 'You must have the variables OTRS_URL, OTRS_USER and OTRS_PASSWORD set in the environment or an .env file' )
+			->addOption( 'owner', 'i', InputOption::VALUE_REQUIRED, 'OTRS Owner id', 27 )
```
First and second `addOption()` parameters are long and short option name, `i` is a strange alias for `owner`.

wiese commented on README.md:
```diff
@@ -1,20 +1,44 @@
+
+Run the command
+
+    php otrs_tool.php update-ticket -owner 27 change_tickets.csv
```
This should probably read `--owner`.

wiese commented on composer.json:
```diff
@@ -19,7 +19,9 @@
     "php": ">=7.1",
     "symfony/console": "^3.2",
-    "vlucas/phpdotenv": "^2.4"
+    "vlucas/phpdotenv": "^2.4",
```
Just recently read [a thread on symfony's dotenv](https://github.com/symfony/dotenv) vs vlucas's. Was this considered?
